src/connection.ts

```diff
@@ -2055,7 +2055,14 @@ class Connection extends EventEmitter {
    * @private
    */
   connectTimeout() {
-    const message = `Failed to connect to ${this.config.server}${this.config.options.port ? `:${this.config.options.port}` : `\\${this.config.options.instanceName}`} in ${this.config.options.connectTimeout}ms`;
```

Comment:
We used to directly create the message from the connection config host and port/instance. However, if there is a rerouting happened, we will be fail to connection to the rerouted host instead of the host from the connection config. It kind make sense in our case since we just let the user know that the host you indicated in the connection config is not connectable due to an error. Since we try to provide more detail on the routing server here, I think the message will be "fail to connect to the routing server (redirected from the orginal server)  like the one in the change.